File: java/src/test/java/com/wgtwo/auth/sample/Sample.java
package com.wgtwo.auth.sample;

import com.wgtwo.api.v1.sms.SmsProto.SendMessageResponse;
import com.wgtwo.api.v1.sms.SmsProto.SendTextToSubscriberRequest;
import com.wgtwo.api.v1.sms.SmsServiceGrpc;
import com.wgtwo.api.v1.sms.SmsServiceGrpc.SmsServiceBlockingStub;
import com.wgtwo.auth.ClientCredentialSource;
import com.wgtwo.auth.WgtwoAuth;
import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;

public class Sample {
  private static final String clientId = System.getenv("CLIENT_ID");
  private static final String clientSecret = System.getenv("CLIENT_SECRET");
  private static final WgtwoAuth wgtwoAuth = WgtwoAuth.builder(clientId, clientSecret).build();
  private static final ClientCredentialSource clientCredentialSource =
      wgtwoAuth.clientCredentials.newTokenSource("sms.text:send_to_subscriber");

  public static void main(String[] args) {
    ManagedChannel channel = ManagedChannelBuilder.forTarget("api.wgtwo.com:443").build();
    SmsServiceBlockingStub stub =
        SmsServiceGrpc.newBlockingStub(channel)
            .withCallCredentials(clientCredentialSource.callCredentials());

    SendTextToSubscriberRequest request =
        SendTextToSubscriberRequest.newBuilder()
            .setContent("Test")
            .setFromAddress("My Product")
            .setToSubscriber("4799990000")
            .build();
    SendMessageResponse response = stub.sendTextToSubscriber(request);
    System.out.println("Send status: " + response.getStatus());

    channel.shutdownNow();
  }
}

File: java/src/test/kotlin/auth/Sample.kt
package auth

import com.wgtwo.api.v1.sms.SmsProto.SendTextToSubscriberRequest
import com.wgtwo.api.v1.sms.SmsServiceGrpc
import com.wgtwo.auth.WgtwoAuth
import io.grpc.ManagedChannelBuilder

class Sample {
    private val clientId = System.getenv("CLIENT_ID")
    private val clientSecret = System.getenv("CLIENT_SECRET")
    private val wgtwoAuth = WgtwoAuth.builder(clientId, clientSecret).build()
    private val clientCredentialSource = wgtwoAuth.clientCredentialSource("sms.text:send_to_subscriber")
    private val callCredentials = clientCredentialSource.callCredentials()

    private val channel = ManagedChannelBuilder.forTarget("api.wgtwo.com:443").build()
    private val stub = SmsServiceGrpc.newBlockingStub(channel)
        .withCallCredentials(callCredentials)

    fun sendSms() {
        val request = SendTextToSubscriberRequest.newBuilder().apply {
            content = "Test"
            fromAddress = "My Product"
            toSubscriber = "4799990000"
        }.build()
        val response = stub.sendTextToSubscriber(request)
    }
}
